docling_core/types/doc/document.py

```diff
@@ -598,10 +602,79 @@ class NodeItem(BaseModel):
 
     model_config = ConfigDict(extra="forbid")
 
-    def get_ref(self):
+    def get_ref(self) -> RefItem:
         """get_ref."""
         return RefItem(cref=self.self_ref)
 
+    def get_parent_ref(
```

Comment:
If that's only to be used by our implementation, instead of extending the public API, I would just add this as an internal helper function on the module level.

Reply:
done